execution.bs

```diff
@@ -4108,28 +4111,219 @@ namespace std::execution {
+    set_error_t(ErrorType)
+    set_stopped_t()</pre>
+
+: completion-signal operation
+:: A valid invocation of a completion function.
+
+: completion scheduler
+:: A scheduler obtainable from a sender's environment that, when present,
+    denotes the scheduler upon which the sender will invoke a completion
+    signal operation. Each completion function may have a different completion
+    scheduler for a particular sender.
+
+: sender algorithm
+:: A function that either takes or returns a sender.
+
+: sender factory
+:: A sender algorithm that takes non-senders as arguments and that returns
+    a sender.
+
+: sender adaptor
+:: A sender algorithm that takes one or more senders and a (possibly empty)
+    set of additional arguments, and that returns a sender.
+
+: sender consumer
+:: A sender algorithm that takes one or more senders and a (possibly empty)
+    set of additional arguments, and that returns a non-sender or `void`.
+
+### Async function protocol <b>[async.protocol]</b> ### {#spec-execution-async.protocol}
+
+The <dfn export=true>async function protocol</dfn> that asynchronous algorithms must
+adhere to.
+
+*  A scheduler may become invalid when the lifetime of the execution resource
+    to which it refers ends, or when the execution resource itself becomes
+    invalid.
+
+* The validity of a sender returned by `schedule(sched)` does not depend on the 
+    lifetime of the scheduler `sched`. It may, however, become invalid when the
+    scheculer that produced it becomes invalid. 
+
+* An operation state obtained by calling `connect` on the result of
+    `scheduler(sched)` may become invalid when the scheduler `sched` becomes
+    invalid.
+
+* The validity of an operation state returned by `connect(snd, rcv)` does not
+    depend on the lifetime of `snd` or `rcv`.
+
+* An operation state shall not be moved or copied once `start()` has begun
+    executing. <span class="wg21note">Operation states are not required to
+    be movable.</span>
+
+* Once `start()` has begun executing, an operation state shall remain valid
+    until a completion-signal operation has begun executing.
+
+* The destructor of an operation state shall be called synchronously, directly
+    or indirectly, from within the execution of the completion-signal operation,
+    or after the completion-signal operation has finished executing.
+
+* A completion-signal operation shall not be called before `start()` has begun
+    executing.
+
+* The receiver to which a sender is connected shall not be copied or moved after
+    `start()` is called on the operation state.
+
+* Once `start()` has been called on an operation state, exactly one of the
+    completion-signal operations shall begin executing on the receiver before
+    the receiver is destroyed. <span class="wg21note">Customarily, the receiver
+    releases resources in its completion-signal operations. This can involve
+    destroying the operation state, which often owns the receiver itself.</note>
```

Comment:
The receiver's completion signal operation is responsible for destroying the operation state, or causing it to be destroyed.

if you are a sender, you can't touch anything after set_value is called. if you are a sender, you will not be destroyed before set_value begins executing. 

Lifetime of the operation state ends, but the storage duration has not ended.